File: jeonghye-choi/42748.js
//---------------------------------------------------------------------------
// Title:         프로그래머스 정렬 K번째수
// Author:        jeonghye-choi
// Last Updated:  2022.2.14
// Link:          https://programmers.co.kr/learn/courses/30/lessons/42748
//---------------------------------------------------------------------------

function solution(array, commands) {
  return commands.map(
    (command) =>
      array.slice(command[0] - 1, command[1]).sort((a, b) => a - b)[
        command[2] - 1
      ]
  );
}

File: pers0n4/42748.py
# K번째수
# https://programmers.co.kr/learn/courses/30/lessons/42748
def solution(array: list[int], commands: list[list[int]]):
    return list(
        map(
            lambda x: sorted(array[x[0] - 1 : x[1]])[x[2] - 1],
            commands,
        )
    )
